### PR TITLE
table rearranging.

### PR DESCRIPTION
**Pull request by HorGra**

### CODE REVIEW

File: persister/src/main/java/com/kh/lolstats/Persister.java [hunk 1]
```diff
@@ -43,6 +44,17 @@ public Persister(String url, String user, String password) throws SQLException,
         initialiseMatchIds();
     }
 
+    private Set<String> getChampionIds() throws FileNotFoundException {
+        Scanner scanner = new Scanner(new File("/home/hrn/IdeaProjects/LoL/persister/src/resources/champions"));
```
Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: Make it read all the champ ids from a CHAMPIONS table

File: persister/src/main/java/com/kh/lolstats/Persister.java [hunk 2]
```diff
@@ -27,14 +29,13 @@
     private String url;
     private String user;
     private String password;
-    private PreparedStatement preparedStatementStatsWinner;
-    private PreparedStatement preparedStatementStatsLoser;
+    private Map<String, Map<Integer, Map<Boolean, PreparedStatement>>> preparedStatementsStats = new HashMap<>();
```
Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: A champion id is an integer, there's no reason to change it to a string

File: persister/src/main/java/com/kh/lolstats/Persister.java [hunk 3]
```diff
@@ -259,29 +289,12 @@ public long getLatestPatchDate() throws SQLException, ParseException {
         }
     }
 
+    //need to close all statements
```
Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: Make it loop through the statements map and close everything and clear the map out.

File: persister/src/main/java/com/kh/lolstats/Persister.java [hunk 4]
```diff
@@ -122,6 +145,7 @@ public void onMatch(JsonNode matchDetails) throws InterruptedException, SQLExcep
 
     private void persist(ArrayList<Match> matches) throws SQLException, ClassNotFoundException {
         int rows = 0;
+        ArrayList<PreparedStatement> statementsToExecute = new ArrayList<>();
```
Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: No need to recreate this list every time, you can reuse it between runs of persist() by making it a field.

Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: But when you do that make sure to clear it out after you're done using it.

File: persister/src/main/java/com/kh/lolstats/Persister.java [hunk 5]
```diff
@@ -15,7 +17,7 @@
 import java.util.concurrent.ConcurrentHashMap;
 
 public class Persister implements Runnable {
```
Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: Currently there's no benefit to persister and parser being in separate projects... they're so dependant on each other that you should either reconstruct the code such that there are logical boundaries or put them in the same project. There are things in here that the persisted doesn't need but still provides as a service to the parser, like the API keys. You should have an intermediate service that provides them. Logically, a persister is supposed to persist and nothing else.

File: persister/src/main/java/com/kh/lolstats/Persister.java (hunk 1 above)
Comment (HorGra) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: done

File: persister/src/main/java/com/kh/lolstats/Persister.java (hunk 2 above)
Comment (HorGra) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: done

File: persister/src/main/java/com/kh/lolstats/Persister.java (hunk 3 above)
Comment (HorGra) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: done

File: persister/src/main/java/com/kh/lolstats/Persister.java (hunk 4 above)
Comment (HorGra) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: done

**Pull request merged**